lib/public/Share/IShareProviderSupportsAccept.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+namespace OCP\Share;
+
+/**
+ * Interface IShareProviderSupportsAccept
+ *
+ * This interface allows to define IShareProvider that can handle the `acceptShare` method,
+ * which is available since Nextcloud 17.
+ *
+ * @since 30.0.0
+ */
+interface IShareProviderSupportsAccept {
```

Comment:
Could also:
```suggestion
interface IShareProviderSupportsAccept extends IShareProvider {
```

but that might only work atm while it's still linear?

Reply:
We often just add new interfaces so you can extend both or just one.
I am fine with both, what do you prefer?

Reply:
> I'd say as you can not be one without the other it makes sense to extend?

Makes sense and the `IShareProviderWithNotification` was implemented similar, so :+1: